internal/apps/dop/providers/project_report/calculate.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package project_report
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	orgpb "github.com/erda-project/erda-proto-go/core/org/pb"
+	"github.com/erda-project/erda/apistructs"
+)
+
+func (p *provider) refreshBasicIterations() error {
+	_, orgs, err := p.Org.ListOrgs(context.Background(), []int64{}, &orgpb.ListOrgRequest{
+		PageNo:   1,
+		PageSize: 9999,
+	}, true)
+	if err != nil {
+		p.Log.Errorf("failed to get all orgs, err: %v", err)
+		return err
+	}
+	for i := range orgs {
+		p.orgSet.Set(orgs[i].ID, orgs[i])
+	}
+
+	projects, err := p.bdl.GetAllProjects()
+	if err != nil {
+		p.Log.Errorf("failed to get all projects, err: %v", err)
+		return err
+	}
+	for i := range projects {
+		p.projectSet.Set(projects[i].ID, &projects[i])
+	}
+
+	for _, projectDto := range projects {
+		orgDto := p.orgSet.Get(projectDto.OrgID)
+		if orgDto == nil {
+			p.Log.Errorf("failed to find org for project: %s, projectID: %d, orgID: %d",
+				projectDto.Name, projectDto.ID, projectDto.OrgID)
+			continue
+		}
+		iterations, err := p.bdl.ListProjectIterations(apistructs.IterationPagingRequest{
```

Comment:
why not ListAllProjectIterations at once?

Reply:
adjusted